modules/end-user-guide/pages/git-credential-store.adoc

```diff
@@ -1,25 +1,89 @@
 
 .Procedure
-. Prepare Git credential file in the link:https://git-scm.com/docs/git-credential-store#_storage_format[Storage format].
-. Encode content of the file to the base64 format.
-. Create a new {platforms-name} secret in the {platforms-name} {orch-namespace} where a {prod-short} workspace will be created.
+. In your home directory, locate and open your `.git-credentials` file if you already have it. Alternatively, if you do not have this file, save a new `.git-credentials` file, using the link:https://git-scm.com/docs/git-credential-store#_storage_format[Git credentials storage format]. Each credential is stored on its own line in the file:
+
++
+[subs="+quotes,+attributes,+macros"]
+----
+https://__<username>__:__<token>__@__<git_server_hostname>__
+----
+
++
+.A line in a `.git-credentials` file
+====
+`https://trailblazer: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@github.com`
+====
 
-* The labels of the secret that is about to be created must match the set of labels configured in `che.workspace.provision.secret.labels` property of {prod-short}. The default labels are:
+. Select credentials from your `.git-credentials` file for the Secret. Encode the selected credentials to Base64 for the next step.
++
+[TIP]
+====
+Linux CLI users can print a Base64-encoded string to standard output by running `base64` as needed:
+
+* To encode all lines in the file:
+** `$ cat .git-credentials | base64 | tr -d '\n'`
+
+* To encode a selected line:
+** `$ echo -n '__<copied_and_pasted_line_from_.git-credentials>__' | base64`
+====
+
+. Create a new {platforms-name} secret in the user's {orch-namespace}.
++
+[source,yaml,subs="+quotes,+attributes,+macros"]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: git-credentials-secret
+  labels:
+    controller.devfile.io/git-credential: 'true' <1>
+    controller.devfile.io/watch-secret: 'true'
+  annotations:
+    controller.devfile.io/mount-path: /etc/secret <2>
+data:
+  credentials: __<Base64_content_of_.git-credentials>__ <3>
+----
++
+<1> The `controller.devfile.io/git-credential` label marks the Secret as containing Git credentials. All Git credential Secrets in your user {orch-namespace} will be merged into a single Secret that will be mounted to the {devworkspace} containers. All of your original Secrets remain intact.
+<2> You can specify a custom absolute path to which the merged Git credentials Secret are to be mounted as a file in the {devworkspace} containers. For example, if the mount path is set to `/etc/secret`, the merged Git credentials store will be mounted at `/etc/secret/credentials`. All Git credentials secrets in your user {orch-namespace} must be set to the same mount path. The mount path can be set to an arbitrary path since the mount path will be automatically set in the Git configuration file configured at `/etc/gitconfig`.
+<3> The Git credentials store content encoded with `base64`.
```

Comment:
```suggestion
<1> The `controller.devfile.io/git-credential` label marks the Secret as containing Git credentials.
<2> A custom absolute path in the {devworkspace} containers. The Secret is mounted as the `credentials` file at this path. The default path is `/`.
<3> The selected content from `.git-credentials` that you encoded to Base64 in the previous step.
+
[TIP]
====
You can create and apply multiple Git credentials Secrets in your user {orch-namespace}. All of them will be copied into one Secret that will be mounted to the {devworkspace} containers. For example, if you set the mount path to `/etc/secret`, then the one Secret with all of your Git credentials will be mounted at `/etc/secret/credentials`. You must set all Git credentials Secrets in your user {orch-namespace} to the same mount path. You can set the mount path to an arbitrary path because the mount path will be automatically set in the Git configuration file configured at `/etc/gitconfig`.
====
```